.sandbox/provision/lifecycle-asg/on-create.sh

```diff
@@ -0,0 +1,50 @@
+#!/bin/bash
+
+# : ${ASG_NAME:=test-windows-provision}
+# : ${AVAILABILITY_ZONE:=us-east-2a}
+# : ${EC2_SSH_KEY_FILE:=/run/sandbox/fs/secrets/shared/sandbox-shared.pem}
+: ${VOLUME_SIZE:=10}
+: ${MAX_RETRIES:=10}
+
+set -ex
```

Comment:
Be careful, `set -e` will abort on any failures before exit code check (`[[ $? -eq 0 ]]`). Please revisit the error handling and make it consistent.

Reply:
I think that is what we want in our checkintgs. Even though the condition check might not trigger before its exiting, I think it is still acceptable as long as the exited statement print out some meaningful info.